src/query/service/src/table_functions/srf/generate_series.rs

```diff
@@ -0,0 +1,381 @@
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn get_table_info(&self) -> &TableInfo {
+        &self.table_info
+    }
+
+    async fn read_partitions(
+        &self,
+        _: Arc<dyn TableContext>,
+        _: Option<PushDownInfo>,
+    ) -> Result<(PartStatistics, Partitions)> {
+        // dummy statistics
+        Ok((PartStatistics::default_exact(), Partitions::default()))
+    }
+
+    fn table_args(&self) -> Option<TableArgs> {
+        Some(TableArgs::new_positioned(vec![
+            self.start.clone(),
+            self.end.clone(),
+            self.step.clone(),
+        ]))
+    }
+
+    fn read_data(
+        &self,
+        ctx: Arc<dyn TableContext>,
+        _plan: &DataSourcePlan,
+        pipeline: &mut Pipeline,
+    ) -> Result<()> {
+        pipeline.add_source(
+            |output| {
+                GenerateSeriesSource::create(
+                    ctx.clone(),
+                    output,
+                    self.start.clone(),
+                    self.end.clone(),
+                    self.step.clone(),
+                )
+            },
+            1,
+        )?;
+        Ok(())
+    }
+}
+
+struct GenerateSeriesSource {
+    finished: bool,
+    start: Scalar,
+    end: Scalar,
+    step: Scalar,
+}
+
+impl GenerateSeriesSource {
+    pub fn create(
+        ctx: Arc<dyn TableContext>,
+        output: Arc<OutputPort>,
+        start: Scalar,
+        end: Scalar,
+        step: Scalar,
+    ) -> Result<ProcessorPtr> {
+        AsyncSourcer::create(ctx.clone(), output, GenerateSeriesSource {
+            start,
+            end,
+            step,
+            finished: false,
+        })
+    }
+}
+
+fn range<T>(start: T, end: T, step: T) -> Vec<T>
+where T: std::ops::Add<Output = T> + std::ops::Sub<Output = T> + PartialOrd + Copy + Default {
+    let mut result = Vec::new();
+    if step == T::default() {
+        return result;
+    }
+    let increase_sign = end >= start;
+    let step_sign = step > T::default();
+    if increase_sign != step_sign {
+        return result;
+    }
+    let mut current = start;
+    if increase_sign {
+        while current <= end {
+            result.push(current);
+            current = current + step;
+        }
+    } else {
+        while current >= end {
+            result.push(current);
+            current = current + step;
+        }
+    }
+    result
+}
+
+#[async_trait::async_trait]
+impl AsyncSource for GenerateSeriesSource {
+    const NAME: &'static str = "GenerateSeriesSourceTransform";
+
+    #[async_trait::unboxed_simple]
+    async fn generate(&mut self) -> Result<Option<DataBlock>> {
+        if self.finished {
+            return Ok(None);
+        }
+        let have_float = number_scalars_have_float(vec![
+            self.start.clone(),
+            self.end.clone(),
+            self.step.clone(),
+        ]);
+        if have_float {
+            let dest_type = DataType::Number(NumberDataType::Float64);
```

Comment:
1. better to have a `macro` to make the codes look better.
2. `date`, `timestamp` should be covered.